src/main/php/lang/XPClass.class.php

```diff
@@ -63,24 +63,47 @@ static function __static() { }
    * Constructor
    *
    * @param   var ref either a class name, a ReflectionClass instance or an object
-   * @throws  lang.IllegalStateException
    */
   public function __construct($ref) {
     if ($ref instanceof \ReflectionClass) {
-      $this->_reflect= $ref;
       $this->_class= $ref->getName();
     } else if (is_object($ref)) {
-      $this->_reflect= new \ReflectionClass($ref);
       $this->_class= get_class($ref);
     } else {
+      $this->_class= (string)$ref;
+    }
+    parent::__construct(\xp::nameOf($this->_class), null);
+  }
+
+  /**
+   * Overload member access, retaining BC for public _reflect member.
+   *
+   * @param  string $name
+   * @return var
+   */
+  public function __get($name) {
```

Comment:
This should be deprecated in one of the upcoming major releases, but no later than XP7.0